extensions/aws/s3/s3-core/src/main/java/org/eclipse/dataspaceconnector/aws/s3/core/AwsClientProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.aws.s3.core;
+
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.SecretToken;
+import software.amazon.awssdk.services.iam.IamAsyncClient;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.sts.StsAsyncClient;
+
+public interface AwsClientProvider {
```

Comment:
A bit of a nit, but for consistency it would be good to add doc on when to use the client